src/mod_blocking.erl

```diff
@@ -27,27 +31,33 @@
 
 -type listitem() :: #listitem{}.
 
-start(Host, _Opts) ->
-    ejabberd_hooks:add(hooks(Host)).
+start(HostType, _Opts) ->
+    ejabberd_hooks:add(hooks(HostType)).
+
+stop(HostType) ->
+    ejabberd_hooks:delete(hooks(HostType)).
 
-stop(Host) ->
-    ejabberd_hooks:delete(hooks(Host)).
+deps(_HostType, Opts) ->
+    [{mod_privacy, Opts, hard}].
```

Comment:
Some tests, e.g. `mod_global_distrib_SUITE` stop `mod_privacy` without stopping `mod_blocking` in line 196:

`ModulesToStop = [mod_offline, mod_privacy, mod_roster, mod_last],`

After that, modules are started with `gen_mod_deps`, which does not start `mod_privacy` as it is a dependency of `mod_blocking`. This makes big tests fail.

To find this I needed quite a bit of debugging. Please make sure it is all consistent now, there can be more such places in the tests.